pkg/node/client.go

```diff
@@ -18,9 +19,19 @@ var NodeClientErr = errs.Class("node client error")
 
 // NewNodeClient instantiates a node client
 func NewNodeClient(self proto.Node) (Client, error) {
+	ca, err := provider.NewCA(context.Background(), 12, 4)
+	if err != nil {
+		return nil, err
+	}
+	identity, err := ca.NewIdentity()
```

Comment:
more importantly, you should be taking an identity as an argument and not making a new one.